app/services/uri_service.rb

```diff
@@ -5,7 +5,7 @@ def initialize(uri)
     return nil if uri.nil?
 
     begin 
-      @uri = URI.parse(URI.encode(uri))
+      @uri = URI.parse(uri)
```

Comment:
this is no longer necessary because the `clean` stuff in config is handling it, right?

Reply:
Correct. We were actually misusing `URI.encode` before - that function is meant only for encoding form data or special characters in paths (it does the special-character-to-`%XX` conversion) - and we were using it to encode the entire URL, including the host. I believe it got put in there because `URI.parse` would crash if you gave it a bad URL; calling `encode` allows those bad URLs to fail silently...